dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -0,0 +1,301 @@
+import com.dotmarketing.portlets.links.model.Link;
+import com.dotmarketing.portlets.links.model.Link.LinkType;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.RegExMatch;
+import com.dotmarketing.util.UtilMethods;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.velocity.tools.view.context.ViewContext;
+import org.apache.velocity.tools.view.tools.ViewTool;
+
+import com.liferay.portal.model.User;
+
+public class NavTool implements ViewTool {
+
+    private Host currenthost = null;
+    private static User systemUser = null;
+    private HttpServletRequest request = null;
+    private long currentLanguage = 0;
+    private ViewContext context;
+    static {
+
+        try {
+            systemUser = APILocator.getUserAPI()
+                .getSystemUser();
+        } catch (DotDataException e) {
+            Logger.error(NavTool.class, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void init(Object initData) {
+        context = (ViewContext) initData;
+        try {
+            this.request = context.getRequest();
+            this.currenthost = WebAPILocator.getHostWebAPI()
+                .getCurrentHost(context.getRequest());
+            this.currentLanguage = WebAPILocator.getLanguageWebAPI()
+                .getLanguage(this.request)
+                .getId();
+        } catch (Exception e) {
+            Logger.warn(this, e.getMessage(), e);
+        }
+    }
+
+    protected NavResult getNav(Host host, String path) throws DotDataException, DotSecurityException {
+        return getNav(host, path, this.currentLanguage, this.systemUser);
+    }
+
+    protected NavResultHydrated getNav(Host host, String path, long languageId, User systemUserParam)
+            throws DotDataException, DotSecurityException {
+
+        if (path != null && path.contains(".")) {
+            path = path.substring(0, path.lastIndexOf("/"));
+        }
+
+        Folder folder = !path.equals("/") ? APILocator.getFolderAPI()
+            .findFolderByPath(path, host, systemUserParam, true)
+                : APILocator.getFolderAPI()
+                    .findSystemFolder();
+        if (folder == null || !UtilMethods.isSet(folder.getIdentifier())) {
+            return null;
+        }
+        NavResult result = CacheLocator.getNavToolCache()
+            .getNav(host.getIdentifier(), folder.getInode(), languageId);
+        if (result != null) {
+
+            return new NavResultHydrated(result, this.context);
+        } else {
+            String parentId;
+            if (!folder.getInode()
+                .equals(FolderAPI.SYSTEM_FOLDER)) {
+                Identifier ident = APILocator.getIdentifierAPI()
+                    .find(folder);
+                parentId = ident.getParentPath()
+                    .equals("/") ? FolderAPI.SYSTEM_FOLDER
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "/" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1132)